routing/datasource.go

```diff
@@ -529,22 +525,32 @@ func mapPredicates(cps []PredicateSpec) map[string]PredicateSpec {
 // processes a set of route definitions for the routing table
 func processRouteDefs(o *Options, defs []*eskip.Route) (routes []*Route, invalidDefs []*eskip.Route) {
 	cpm := mapPredicates(o.Predicates)
+	reasonCounts := make(map[string]int)
+
 	for _, def := range defs {
 		route, err := processRouteDef(o, cpm, def)
 		if err == nil {
 			routes = append(routes, route)
 		} else {
 			invalidDefs = append(invalidDefs, def)
-			o.Log.Errorf("failed to process route %s: %v", def.Id, err)
+			if o.Log != nil {
```

Comment:
Why do we add nil check here? It creates confusion signalling that log can be nil while it was never nil before.